### PR TITLE
Give more flexibility to requeueing parameters in the config api

### PR DESCRIPTION
**Pull request by tenzen-y**
#### What type of PR is this?

/kind feature
/kind api-change

#### What this PR does / why we need it:
I added the `RequeueingStrategy` to the config API based on https://github.com/kubernetes-sigs/kueue/tree/f6d6adee77cca1cbf312bc65302025a9aa16129a/keps/1282-pods-ready-requeue-strategy.

Note that the existing `RequeuingTimestamp` API hasn't yet been released. So, we can break this API and I don't think that the release-note is needed.

#### Which issue(s) this PR fixes:

Part-of: #1282 

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: go.mod
Comment (tenzen-y) [inline go.mod:1]: I'm not sure why this commit is included. I'm cleaning up now.
/hold

Comment (tenzen-y) [reply go.mod:1]: done.
/hold cancel

**Review by alculquicondor — COMMENTED** [review]
/lgtm

**Pull request merged**